src/openassetio-python/setup.py

```diff
@@ -39,10 +40,22 @@ def build_extension(self, _ext: Extension):
 
         @param _ext: Extension to build (we only have one, so ignored).
         """
+        cmake_project_path = pathlib.Path("../..")
```

Comment:
[question] Could we use `os.path.abspath(__file__)` or similar to avoid `cwd` sensitive paths?

Reply:
IIRC `__file__` won't work because `build_ext` seems to be ultimately called from an `exec` statement.  It's all very weird.  I'll double-check though.

Reply:
I tried a quick print and it did give me `setup.py`, but I hadn't tried with `abspath`

Reply:
Confirmed, you get `setup.py`, but `__file__` is normally the absolute path to the file, not just the file name. `abspath` will then prefix the cwd to that.  So it's kinda equivalent to just using `.`.

I'll see if there's a better option available, though.

I was a little wrong about the `exec` statement thing - it is ultimately called from `exec`, but that's not the reason we can't get the path of `setup.py` from inside `build_ext`.  It is ultimately running `build_ext` as if under `exec` (i.e. like executing a string), but I'd have to spend a bit more time debugging to really get to the bottom of it.

Reply:
Don't worry for now if its not easily solvable. I was just curious if it was an option. I was more confused that it _is_ working when you do `python -m pip install src/openassetio-python` if that is making a wheel!

Reply:
Yeah, it's a really long winding process, e.g. with a breakpoint in `built_ext` I get
```
sys.argv
['/home/dave/workspace/cloud/assetapi/OpenAssetIO/.venv/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py', 'bdist_wheel', '--dist-dir', '/tmp/pip-wheel-eed3kpwi/tmpxuecirt8']
```
and then early on in the stack trace there is an `exec` call adding yet more confusion...

Anyway, I tried all sorts in the debugger to see if there was some variable available somewhere that could give the directory of `setup.py`, but can't find anything.

Is it OK as it is, or does it need some clarification? - could be explicit and use `os.cwd`, or could add a code comment.

Reply:
If I've understood what you were saying above re: "not needing to be in the dir" is only for advanced non-standard wheel builds, then I think it'd be fine as is - especially if `python -m pip install` and others do the right thing for you? Sorry for the noise here. Good job exhausting options!